target/linux/layerscape/patches-5.4/820-usb-0018-MLK-18794-1-usb-host-xhci-add-.bus_suspend-override.patch

```diff
@@ -29,17 +29,17 @@ Signed-off-by: Peter Chen <peter.chen@nxp.com>
  		if (over->start)
  			drv->start = over->start;
 +		if (over->bus_suspend)
-+			drv->bus_suspend = over->bus_suspend;
- 	}
- }
- EXPORT_SYMBOL_GPL(xhci_init_driver);
++			drv->->bus_suspend = over->bus_suspend;
```

Comment:
How did you manage to get this error in?

I will fix it myself now, but I still am startled from how strange manual modification are introduced in places where I'd never have expected them.

Reply:
Well, I started over.
```
...
Applying patch platform/820-usb-0018-MLK-18794-1-usb-host-xhci-add-.bus_suspend-override.patch
patching file drivers/usb/host/xhci.c
Hunk #1 FAILED at 5381.
1 out of 1 hunk FAILED -- rejects in file drivers/usb/host/xhci.c
patching file drivers/usb/host/xhci.h
Hunk #1 FAILED at 1912.
1 out of 1 hunk FAILED -- rejects in file drivers/usb/host/xhci.h
Patch platform/820-usb-0018-MLK-18794-1-usb-host-xhci-add-.bus_suspend-override.patch does not apply (enforce with -f)
make[3]: *** [Makefile:22: /scratch/union/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-5.4.97/.quilt_checked] Error 1
make[3]: Leaving directory '/scratch/union/target/linux/layerscape'
make[2]: *** [Makefile:11: refresh] Error 2
make[2]: Leaving directory '/scratch/union/target/linux'
time: target/linux/refresh#13.22#3.46#14.67
    ERROR: target/linux failed to build.
make[1]: *** [target/Makefile:25: target/linux/refresh] Error 1
make[1]: Leaving directory '/scratch/union'
make: *** [/scratch/union/include/toplevel.mk:230: target/linux/refresh] Error 2
```

So:
```
% cd /scratch/union/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-5.4.97
% quilt push -f
Applying patch platform/820-usb-0018-MLK-18794-1-usb-host-xhci-add-.bus_suspend-override.patch
patching file drivers/usb/host/xhci.c
Hunk #1 FAILED at 5381.
1 out of 1 hunk FAILED -- saving rejects to file drivers/usb/host/xhci.c.rej
patching file drivers/usb/host/xhci.h
Hunk #1 FAILED at 1912.
1 out of 1 hunk FAILED -- saving rejects to file drivers/usb/host/xhci.h.rej
Applied patch platform/820-usb-0018-MLK-18794-1-usb-host-xhci-add-.bus_suspend-override.patch (forced; needs refresh)
```

* [Here](https://gist.github.com/graysky2/40ce688f338adb4380e22f23251fad5e) is `drivers/usb/host/xhci.c`
* [Here](https://gist.github.com/graysky2/b8c432cee8fed201ba713307c6b63cb2) is `drivers/usb/host/xhci.c.rej`
* After I manually added those two lines, [here](https://gist.github.com/graysky2/d3afc6a5bf13f0d69cbcc8c136fc25bd) is `drivers/usb/host/xhci.c` (edited)

I added the single line to `drivers/usb/host/xhci.h` based on its fail and then ran `quilt refresh` then in the original term, resumed running `./update_kernel.sh -v -u 5.4 5.4.97`

Reply:
Looks like you manually edited the "->->" in ...

Reply:
What I just don't understand is how you got a manual change in that line at all, since there would have been no reason to touch it at all.

Reply:
@adschm - I am to blame for the double "->" must have been introduced when I manually merged.